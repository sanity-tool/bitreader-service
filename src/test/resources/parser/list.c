struct List {
    struct List *next;
} *PLIST;

static void switchToNext() {
    PLIST = PLIST->next;
}