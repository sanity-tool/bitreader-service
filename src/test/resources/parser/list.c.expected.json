{
  "functions" : [ {
    "blocks" : [ {
      "instructions" : [ {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMGetElementPtr",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        }, {
          "kind" : "LLVMConstantIntValueKind"
        } ]
      }, {
        "kind" : "LLVMLoad",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        } ]
      }, {
        "kind" : "LLVMStore",
        "operands" : [ {
          "kind" : "LLVMInstructionValueKind"
        }, {
          "kind" : "LLVMGlobalVariableValueKind"
        } ]
      }, {
        "kind" : "LLVMRet",
        "operands" : [ ]
      } ]
    } ],
    "entryBlockIndex" : 0
  } ]
}